### PR TITLE
Update cypress test for custom serving runtime 

### PR DESCRIPTION
**Pull request by ppadti**
Closes: [RHOAIENG-3835](https://issues.redhat.com/browse/RHOAIENG-3835)

## Description
This PR aims to update cypress test for custom serving runtime 

## How Has This Been Tested?
`npm run test`

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by manaswinidas — COMMENTED** [review]
/lgtm

File: frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts [hunk 1]
```diff
@@ -73,38 +92,239 @@ describe('Custom serving runtimes', () => {
...
+        pathname: '/api/templates/',
+      },
+      mockServingRuntimeTemplateK8sResource({}),
+    ).as('createTemplate');
 
     servingRuntimes.findAddButton().click();
+    servingRuntimes.findAppTitle().should('contain', 'Add serving runtime');
 
-    // Create with multi model
-    servingRuntimes.findCreateButton().should('be.disabled');
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts:92]: Seems still valid to check if the button starts off as disabled.

File: frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts [hunk 2]
```diff
@@ -73,38 +92,239 @@ describe('Custom serving runtimes', () => {
...
+        method: 'PATCH',
+        pathname: '/api/templates/opendatahub/template-1',
+      },
+      mockServingRuntimeTemplateK8sResource({}),
+    ).as('editTemplate');
+
     servingRuntimes.getRowById('template-1').find().findKebabAction('Edit').click();
-    servingRuntimes.findAppTitle().should('have.text', 'Edit Multi Platform');
+    cy.url().should('include', '/editServingRuntime/template-1');
+    cy.get('h1').should('contain', 'Edit Multi Platform');
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts:278]: Why remove `servingRuntimes.findAppTitle().should('have.text', 'Edit Multi Platform');` and the use `cy.get` directly?

File: frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/mock-custom-serving-runtime-add.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+apiVersion: serving.kserve.io/v1alpha1
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/mock-custom-serving-runtime-add.yaml:1]: Move these yaml files to `__mocks__/yaml`.

File: frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts [hunk 4]
```diff
@@ -73,38 +92,239 @@ describe('Custom serving runtimes', () => {
...
+
     servingRuntimes.getRowById('template-1').find().findKebabAction('Duplicate').click();
     servingRuntimes.findAppTitle().should('have.text', 'Duplicate serving runtime');
+    cy.url().should('include', '/addServingRuntime');
+
+    servingRuntimes.shouldDisplayAPIProtocolValues([
+      ServingRuntimeAPIProtocol.REST,
+      ServingRuntimeAPIProtocol.GRPC,
+    ]);
+    servingRuntimes.shouldSelectAPIProtocol(ServingRuntimeAPIProtocol.GRPC);
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts:226]: Please rename `shouldSelectAPIProtocol` to `selectAPIProtocol` as the `should` prefix is used only for assertion functions.

File: frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts (hunk 2 above)
Comment (ppadti) [reply frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts:278]: ooh, I'm sorry. it was by mistake. I will update it.

**Pull request merged**